test/functional/test_framework/test_framework.py

```diff
@@ -203,67 +197,60 @@ def main(self):
 
     # Public helper methods. These can be accessed by the subclass test scripts.
 
-    def start_node(self, i, dirname, extra_args=None, rpchost=None, timewait=None, binary=None, stderr=None):
+    def start_node(self, i, dirname, extra_args=[], rpchost=None, timewait=None, binary=None, stderr=None):
```

Comment:
Why not allow `None` for `extra_args`? I'm generally wary of passing in objects as defaults, as if those default objects are ever altered, the altered version will be passed as the default in future calls.

E.g. try the following snippet:
```Python
def foo(x=[]):
    x.append(5)
    print(x)

foo()
foo()
foo()
```
You will get as output
```
[5]
[5, 5]
[5, 5, 5]
```